src/library/library.cpp

```diff
@@ -378,3 +552,248 @@ void Library::slotSetTrackTableRowHeight(int rowHeight) {
     m_iTrackTableRowHeight = rowHeight;
     emit(setTrackTableRowHeight(rowHeight));
 }
+
+void Library::slotSetHoveredFeature(LibraryFeature* pFeature) {
+    m_hoveredFeature = pFeature;
```

Comment:
The m_hoveredFeature does not take ownership to pFeature and it is checked for null before usage so it can simply become a QPointer member. 

However, it is not a good idea to mix shard_ptr and QPointer because from QT 5 it counts references twice. This means you should use QSharedPointer to track the lifetime. 

The best solution from a clean code view would be to use shard_ptr along with weak_ptr or the Qt equivalent. But that requires some changes and is not required from the business logic, because the Library already owns the feature.